app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -0,0 +1,220 @@
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun deleteUserById(
+      userId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(userId)
+        .delete()
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { onFailure(it) }
+  }
+  /*
+  // Technically this keeps track of lastKnownLocation
+  override fun updateUserLocation(
+      userId: String,
+      location: Location,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val locationData =
+        mapOf(
+            "center" to
+                mapOf(
+                    "latitude" to location.center.latitude(),
+                    "longitude" to location.center.longitude()),
+            "topLeft" to
+                location.topLeft?.let {
+                  mapOf("latitude" to it.latitude(), "longitude" to it.longitude())
+                },
+            "topRight" to
+                location.topRight?.let {
+                  mapOf("latitude" to it.latitude(), "longitude" to it.longitude())
+                },
+            "bottomLeft" to
+                location.bottomLeft?.let {
+                  mapOf("latitude" to it.latitude(), "longitude" to it.longitude())
+                },
+            "bottomRight" to
+                location.bottomRight?.let {
+                  mapOf("latitude" to it.latitude(), "longitude" to it.longitude())
+                })
+
+    db.collection(collectionPath)
+        .document(userId)
+        .update("lastKnownLocation", locationData)
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun getLastKnownLocation(
+      userId: String,
+      onSuccess: (Location?) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(userId)
+        .get()
+        .addOnSuccessListener { document ->
+          val user = document.toObject(User::class.java)
+          onSuccess(user?.lastKnownLocation)
+        }
+        .addOnFailureListener { onFailure(it) }
+  } */
+  override fun addFavoriteParking(
+      userId: String,
+      parkingId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    /*db.collection(collectionPath)
+    .document(userId)
+    .get()
+    .addOnSuccessListener { document ->
+      val user = document.toObject(User::class.java)
+      val updatedFavorites = user?.favoriteParkingSpots?.toMutableList() ?: mutableListOf()
+      updatedFavorites.add(parkingId)
+      db.collection(collectionPath)
+          .document(userId)
+          .update("favoriteParkingSpots", updatedFavorites)
+          .addOnSuccessListener { onSuccess() }
+          .addOnFailureListener { onFailure(it) }
+    }
+    .addOnFailureListener { onFailure(it) }
+    */
+    TODO("To fix due to serialization issues")
```

Comment:
We'll need to fix these serialization issues before integrating the ViewModel into a Screen